system/modules/core/models/ArticleModel.php

```diff
@@ -203,6 +203,34 @@ public static function findPublishedByPidAndColumn($intPid, $strColumn, array $a
 
 
 	/**
+	 * Find all published articles with teaser by their parent ID
+	 *
+	 * @param integer $intPid     The page ID
+	 * @param array   $arrOptions An optional options array
+	 *
+	 * @return \Model\Collection|\ArticleModel[]|\ArticleModel|null A collection of models or null if there are no articles in the given column
+	 */
+	public static function findPublishedWithTeaserByPid($intPid, array $arrOptions=array())
+	{
+		$t = static::$strTable;
+		$arrColumns = array("$t.pid=? AND $t.showTeaser=1");
+
+		if (!BE_USER_LOGGED_IN)
```

Comment:
That's what I mean. We just don't know. To my gut feeling, we're changing too much, we should just fix the initial problem and not start adding new behavior.